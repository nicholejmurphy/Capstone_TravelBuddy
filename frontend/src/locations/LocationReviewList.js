import React from "react";
import LocationReviewCard from "./LocationReviewCard";

/** Shows a list of all locations
 *
 * - Recieves reviews list from parent: LcoationDetails
 * - Passes review data to children: LocationReviewCard
 */
function LocationReviewList({ reviews }) {
  return (
    <div className="">
      {reviews ? (
        <div>
          <h4>REVIEWS</h4>
          {reviews.map((r) => (
            <LocationReviewCard key={r.id} review={r} />
          ))}
        </div>
      ) : (
        <div>
          <p>No reviews to show.</p>
        </div>
      )}
    </div>
  );
}

export default LocationReviewList;
