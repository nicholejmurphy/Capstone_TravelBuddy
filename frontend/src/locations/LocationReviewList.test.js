import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import LocationReviewList from "./LocationReviewList";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <LocationReviewList reviews={[]} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot for no reviews", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocationReviewList reviews={[]} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot with reviews", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocationReviewList reviews={[{}]} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
