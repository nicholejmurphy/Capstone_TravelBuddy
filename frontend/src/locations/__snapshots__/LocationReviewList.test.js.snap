// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for no reviews 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div>
      <h4>
        REVIEWS
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot with reviews 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div>
      <h4>
        REVIEWS
      </h4>
      <div
        class=""
      >
        <div>
          <p />
          <p>
            Date: 
          </p>
          <p>
            Rating:  
            <img
              alt="rating icon"
            />
          </p>
          <p>
            Trip type: 
          </p>
          <p />
          <small>
            <a>
              Read More on TripAdvisor
            </a>
          </small>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
